cpp/ql/src/semmle/code/cpp/models/implementations/Allocation.qll

```diff
@@ -215,6 +215,39 @@ class SizelessAllocationFunction extends AllocationFunction {
   }
 }
 
+/**
+ * An `operator new` or `operator new[]` function that may be associated with `new` or
+ * `new[]` expressions.  Note that `new` and `new[]` are not function calls, but these
+ * functions may also be called directly.
+ */
+class OperatorNewAllocationFunction extends AllocationFunction {
+  OperatorNewAllocationFunction() {
+    exists(string name |
+      hasGlobalOrStdName(name) and
```

Comment:
AFAICT from https://en.cppreference.com/w/cpp/memory/new/operator_new, these functions are always in the global namespace, not `std`.

Reply:
Done.